tests/runtests.py

```diff
@@ -0,0 +1,337 @@
+        if cls.config['driver'] == "chromium":
+            chrome_options = ChromiumOptions()
+            chrome_options.add_argument('--ignore-certificate-errors')
+            capabilities = DesiredCapabilities.CHROME
+            capabilities['goog:loggingPrefs'] = {'browser': 'ALL'}
+            if cls.config['headless']:
+                chrome_options.add_argument("--headless")
+            cls.base_driver = webdriver.Chrome(options=chrome_options,
+                                               desired_capabilities=capabilities)
+            cls.second_driver = webdriver.Chrome(options=chrome_options,
+                                                 desired_capabilities=capabilities)
+        cls.base_driver.set_window_size(1366, 768)
+        cls.second_driver.set_window_size(1366, 768)
+
+    @classmethod
+    def tearDownClass(cls):
+        for resource_link in cls.objects_to_delete:
+            try:
+                cls._delete_object(resource_link)
+            except NoSuchElementException:
+                print("Unable to delete resource at: " +
+                      resource_link)
+        cls.second_driver.close()
+        cls.base_driver.close()
+        if cls.failed_test and cls.config['logs']:
+            cmd = subprocess.Popen(['docker-compose', 'logs'],
+                                   universal_newlines=True,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   cwd=cls.root_location)
+            output, _ = map(str, cmd.communicate())
+            print("One of the containers are down!\nOutput:\n" +
+                  output)
+
+    @classmethod
+    def _delete_object(cls, resource_link):
+        """
+        Takes URL for location to delete.
+        """
+        cls.base_driver.get(resource_link)
+        cls.base_driver.find_element_by_class_name('submit-row') \
+                       .find_element_by_class_name('deletelink-box') \
+                       .click()
+        cls.base_driver.find_element_by_xpath('//input[@type="submit"]').click()
+
+    def test_topology_graph(self):
+        path = '/admin/topology/topology'
+        label = 'automated-selenium-test-02'
+        self.login()
+        self.create_network_topology(label)
+        self.action_on_resource(label, path, 'delete_selected')
+        self.assertNotIn('Nodes', self.base_driver.page_source)
+        self.action_on_resource(label, path, 'update_selected')
+        self.action_on_resource(label, path, 'delete_selected')
+        self.assertIn('Nodes', self.base_driver.page_source)
+
+
+    def test_topology_visual(self):
+
+        def _visualize_click():
+            # TODO: Ensure that we open the visualize page,
+            # I have tried waiting for the page to load but
+            # it doesn't work, for some reason I am not able
+            # to find, it doesn't work on the first click something.
```

Comment:
I used this because hack because `visualize` page takes time to load and and the test fails randomly at times.
I tried a couple of tricks that didn't work:
1. `time.sleep()`
2. Using selenium [wait on element](https://stackoverflow.com/a/26567563/6410464)
3. wait for `document.ready`
4. explicitly checking if `visualize.js` is loaded before running tests.
5. using `DOMAIN/topology/topology/ID` page instead.

but none of the tricks worked reliably, I get the same result of race condition and test passing half of the time. 
Unfortunately, I am out of ideas.
Hence I am opening an issue for the same. :smile:

Reply:
Removed this test from this PR, it already has another issue, added these details in that issue.